File: Examples/intel_examples/examples_core_c/dftc/source/error_processing.c
/*******************************************************************************
* Copyright 2011-2020 Intel Corporation.
*
* This software and the related documents are Intel copyrighted  materials,  and
* your use of  them is  governed by the  express license  under which  they were
* provided to you (License).  Unless the License provides otherwise, you may not
* use, modify, copy, publish, distribute,  disclose or transmit this software or
* the related documents without Intel's prior written permission.
*
* This software and the related documents  are provided as  is,  with no express
* or implied  warranties,  other  than those  that are  expressly stated  in the
* License.
*******************************************************************************/

/*
! Content:
! An example of error processing when using DFTI functions.
!
!****************************************************************************/

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <float.h>
#include "mkl_dfti.h"

/* Define the format to printf MKL_LONG values */
#if !defined(MKL_ILP64)
#define LI "%li"
#else
#define LI "%lli"
#endif

static void report_error(MKL_LONG status);

int main(void)
{
    /* Execution status */
    MKL_LONG status = 0;

    DFTI_DESCRIPTOR_HANDLE hand = NULL;

    printf("Example error_processing\n");

    printf("Try to create a DFTI descriptor with misplaced arguments\n");
    status = DftiCreateDescriptor(&hand, DFTI_COMPLEX, DFTI_SINGLE, 1, 1024);
    if (status != DFTI_NO_ERROR) report_error(status);

    if (status == DFTI_NO_ERROR) {
        printf("TEST FAILED\n");
        return 1;
    } else {
        printf("TEST PASSED\n");
        return 0;
    }
}

static void report_error(MKL_LONG status)
{
    printf(" Nonzero status = "LI"\n", status);
    printf(" Check if the status indicates of error\n");
    if (!DftiErrorClass(status, DFTI_NO_ERROR)) {
        printf("   Error: %s\n", DftiErrorMessage(status));
    } else {
        printf("   Not an error: %s\n", DftiErrorMessage(status));
    }
}

File: Examples/intel_examples/examples_core_f/dftf/source/error_processing.f90
!===============================================================================
! Copyright 2011-2020 Intel Corporation.
!
! This software and the related documents are Intel copyrighted  materials,  and
! your use of  them is  governed by the  express license  under which  they were
! provided to you (License).  Unless the License provides otherwise, you may not
! use, modify, copy, publish, distribute,  disclose or transmit this software or
! the related documents without Intel's prior written permission.
!
! This software and the related documents  are provided as  is,  with no express
! or implied  warranties,  other  than those  that are  expressly stated  in the
! License.
!===============================================================================

! Content:
! An example of error processing when using DFTI functions.
!
!*****************************************************************************

program error_processing

  use MKL_DFTI

  ! Execution status
  integer :: status = 0

  ! DFTI descriptor handle
  type(DFTI_DESCRIPTOR), POINTER :: hand

  hand => null()

  print *,"Example error_processing"

  print *,"Try to create a DFTI descriptor with misplaced arguments"
  status = DftiCreateDescriptor(hand, DFTI_COMPLEX, DFTI_DOUBLE, 1, 2)
  if (0 /= status) call report_error(status)

  if (0 == status) then
    print *,"TEST FAILED"
    call exit(1)
  else
    print *,"TEST PASSED"
    call exit(0)
  end if

contains

  subroutine report_error(s)
    integer s

    print '(" Nonzero status = "I0)', s
    print '(" Check if the status indicates of error")'
    if (.not. DftiErrorClass( s, DFTI_NO_ERROR)) then
      print '("  Error: "A)', DftiErrorMessage(s)
    else
      print '("  Not an error: "A)', DftiErrorMessage(s)
    end if
  end subroutine report_error

end program error_processing
